packages/sandpack-core/src/transpiled-module/transpiled-module.ts

```diff
@@ -792,52 +787,18 @@ export class TranspiledModule {
     }: { asUMD?: boolean; force?: boolean; globals?: any } = {},
     initiator?: TranspiledModule
   ) {
+    // empty module
+    if (this.module.path === '/node_modules/empty/index.js') {
+      return {};
+    }
+
     // Just let the browser reload...
     if (manager.isReloading) {
       return {};
     }
 
     if (this.source == null) {
-      if (this.module.path === '/node_modules/empty/index.js') {
-        return {};
-      }
-
-      // TODO: Never return untranspiled modules, as these can always have side-effects...
-      if (
-        this.module.path.startsWith('/node_modules') &&
-        !this.module.path.endsWith('.vue')
-      ) {
-        if (process.env.NODE_ENV === 'development') {
-          console.warn(
-            `[WARN] Sandpack: loading an untranspiled module: ${this.module.path}`
-          );
-        }
-        // This code is probably required as a dynamic require. Since we can
-        // assume that node_modules dynamic requires are only done for node
-        // utilites we will just set the transpiled code according to how
-        // node handles that.
-
-        let code = this.module.path.endsWith('.json')
-          ? `module.exports = JSON.parse(${JSON.stringify(this.module.code)})`
-          : this.module.code;
-
-        code += `\n//# sourceURL=${location.origin}${this.module.path}${
-          this.query ? `?${this.hash}` : ''
-        }`;
-
-        this.source = new ModuleSource(this.module.path, code, null);
-
-        if (initiator) {
-          initiator.dependencies.add(this);
-          this.initiators.add(initiator);
-        }
-      } else {
-        // This scenario only happens when we are in an inconsistent state, the quickest way to solve
-        // this state is to just hard reload everything.
-        manager.clearCache();
-
-        throw new Error(`${this.getId()} hasn't been transpiled yet.`);
-      }
+      throw new Error(`${this.getId()} hasn't been transpiled yet.`);
```

Comment:
I think this might break sandboxes. Not completely sure, but there must be cases where we have a dynamic import or a `import("./index" + i + ".js")` or something.

Reply:
dynamic imports get transpiled? They're promise based. The old dynamic require logic will likely fail though, not sure if we can even support that

Reply:
Ohhhh good point, I was meaning `require()`. Yeah hmm, I think only old sandboxes use it. Which highlights our need to have versioning in the bundler...